godtools/App/Share/Common/URLSession/MobileContentApiAuthSession.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  MobileContentApiAuthSession.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 11/10/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import Foundation
+import RequestOperation
+import Combine
+
+class MobileContentApiAuthSession {
+    
+    let mobileContentAuthTokenRepository: MobileContentAuthTokenRepository
+    let session: URLSession
+    
+    private let requestBuilder: RequestBuilder = RequestBuilder()
+    
+    init(ignoreCacheSession: IgnoreCacheSession, mobileContentAuthTokenRepository: MobileContentAuthTokenRepository) {
+     
+        self.session = ignoreCacheSession.session
+        self.mobileContentAuthTokenRepository = mobileContentAuthTokenRepository
+    }
+    
+    func sendDataRequest(with urlString: String) -> AnyPublisher<Data?, URLResponseError> {
```

Comment:
Hey @rachaelblue I should have made this note in the Jira task, but I would like an additional method that accepts a (URLSession and URLRequest).  The idea here is MobileContentApiAuthSession would mutate the URLRequest by adding the Auth header and then use the provided URLSession to execute the URLRequest.  This would give us more modularity as the URLSession and URLRequest can be created completely separate from MobileContentApiAuthSession.

I think the urlString method is nice as well for convenience, just note that we are locked into using the IgnoreCacheSession and http methods with GET.  Maybe the naming could be a little more specific to that.  ```sendGetRequestIgnoringCache``` or something along those lines.

Reply:
Fixed!

Reply:
to further add to rationale behind this, several of the new protected APIs take POST and PUT requests to submit data to the API. So, something hardcoded to be GET only has limited utility